core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -122,7 +127,18 @@ DeleteFile[] forEntry(ManifestEntry<DataFile> entry) {
     return forDataFile(entry.dataSequenceNumber(), entry.file());
   }
 
+  DeleteFile[] forDataFile(DataFile file) {
```

Comment:
We expose sequence numbers on `DataFile` so there is no need to have `ManifestEntry` anymore.

Reply:
The changes in this file look correct to me. Only problem is dropping the `minSequenceNumber` filter. Wouldn't hurt to keep that.

Reply:
Resolving this thread as all other changes have been merged.